src/string.cr

```diff
@@ -3627,37 +3627,61 @@ class String
   # "Aubergine".rjust(8)   # => "Aubergine"
   # ```
   def rjust(len, char : Char = ' ')
-    just len, char, false
+    just len, char, 1
   end
 
-  private def just(len, char, left)
+  # Adds instances of *char* to left ond right of the string until it is at least size of *len*.
+  #
+  # ```
+  # "Purple".center(8)      # => " Purple "
+  # "Purple".center(8, '-') # => "-Purple-"
+  # "Purple".center(9, '-') # => "-Purple--"
+  # "Aubergine".center(8)   # => "Aubergine"
+  # ```
+  def center(len, char : Char = ' ')
+    just len, char, 0
+  end
+
+  private def just(len, char, justify)
     return self if size >= len
 
     bytes, count = String.char_bytes_and_bytesize(char)
-
-    difference = len - size
-    new_bytesize = bytesize + difference * count
+    padding = (len - size)
+    new_bytesize = bytesize + padding * count
+    case justify
+    when .< 0
+      leftpadding, rightpadding = 0, padding
+    when .> 0
+      leftpadding, rightpadding = padding, 0
+    else
+      leftpadding = padding >> 1
+      rightpadding = padding - leftpadding
+    end
 
     String.new(new_bytesize) do |buffer|
-      if left
-        buffer.copy_from(to_unsafe, bytesize)
-        buffer += bytesize
-      end
-
-      if count == 1
-        Intrinsics.memset(buffer.as(Void*), char.ord.to_u8, difference.to_u32, false)
-        buffer += difference
-      else
-        difference.times do
-          buffer.copy_from(bytes.to_unsafe, count)
-          buffer += count
+      if leftpadding > 0
+        if count == 1
+          Intrinsics.memset(buffer.as(Void*), char.ord.to_u8, leftpadding.to_u32, false)
+          buffer += leftpadding
+        else
+          leftpadding.times do
+            Intrinsics.memcpy(buffer.as(Void*), bytes.to_unsafe.as(Void*), count, false)
+            buffer += count
+          end
         end
       end
-
-      unless left
-        buffer.copy_from(to_unsafe, bytesize)
+      Intrinsics.memcpy(buffer.as(Void*), to_unsafe.as(Void*), bytesize, false)
```

Comment:
I think you might be able to use `Pointer.copy_from`

```suggestion
      buffer.copy_from(to_unsafe, bytesize)
```

Reply:
I see now this is diff noise so this might be outside the scope of this PR.

Reply:
Is this "better" than memcpy ?

Reply:
It's the same as `memcpy`, but shorter.

It's fine to leave this as-is for this PR.